libmikmod/playercode/mplayer.c

```diff
@@ -1157,7 +1157,7 @@ static int DoS3MEffectA(UWORD tick, UWORD flags, MP_CONTROL *a, MODULE *mod, SWO
 	if (tick || mod->patdly2)
 		return 0;
 
-	if (speed > 128)
+	if (speed >= 128)
 		speed -= 128;
```

Comment:
Finally got around to testing all of the formats I mentioned above. Everything that uses the S3M speed effect expects values 0x80-0xFF to map to speeds 128-255.

S3MEFFECTA compatibility:

| Loader | Player | Effect | Values >=0x80? |
|-|-|-|-|
| load_amf.c	| DMP			| 0x01		| values over 0x80 as speed >=128 (MikMod is wrong)
| load_gdm.c	| BWSB			| 0x0F		| values over 0x80 as speed >=128 (both xmp and MikMod are wrong)
| load_imf.c	| IMAGO Orpheus	| 0x01 (1)		| values over 0x80 as speed >=128 (MikMod is wrong, xmp support broken?)
| load_it.c		| Impulse Tracker	| 0x01 (A)		| values over 0x80 as speed >=128 (MikMod is wrong)
| load_med.c	| OctaMED		| 0x09 (09)	| bounded to 0x20
| load_s3m.c	| Scream Tracker	| 0x01 (A)		| values over 0x80 as speed >=128 (MikMod is wrong)

So, IMO, the correct fix is to just remove both of these lines altogether. Some testing on my end suggests doing this will reproduce the behavior I saw in the trackers/players I tested.

Test files used for GDM, IMF, IT, S3M: [S3MEFFECTA.zip](https://github.com/sezero/mikmod/files/6569379/S3MEFFECTA.zip).